frontend/src/app/event/event-page/event-page.component.ts

```diff
@@ -50,34 +80,97 @@ export class EventPageComponent implements OnInit {
   ) {
     // Initialize data from resolvers
     const data = this.route.snapshot.data as {
       profile: Profile;
-      events: Event[];
+      page: PaginatedEvent<Event>;
     };
     this.profile = data.profile;
-    this.events = data.events;
+    this.page = data.page;
+    this.today =
+      this.startDate.setHours(0, 0, 0, 0) == new Date().setHours(0, 0, 0, 0);
 
     // Group events by their dates
-    this.eventsPerDay = eventService.groupEventsByDate(this.events);
+    this.eventsPerDay = eventService.groupEventsByDate(this.page.items);
 
     // Initialize the initially selected event
-    if (data.events.length > 0) {
-      this.selectedEvent = eventFilterPipe.transform(data.events, '')[0];
+    if (data.page.items.length > 0) {
+      this.selectedEvent = this.page.items[0];
     }
+
+    this.searchUpdate
+      .pipe(
+        filter((search: string) => search.length > 2 || search.length == 0),
+        debounceTime(500),
+        distinctUntilChanged()
+      )
+      .subscribe((query) => {
+        this.onSearchBarQueryChange(query);
+      });
   }
 
   /** Runs when the frontend UI loads */
   ngOnInit() {
     // Keep track of the initial width of the browser window
     this.innerWidth = window.innerWidth;
+
+    // Watch current route's query params
+    this.route.queryParams.subscribe((params: Params): void => {
+      this.startDate = params['start_date']
+        ? new Date(Date.parse(params['start_date']))
+        : new Date();
+      this.endDate = params['end_date']
+        ? new Date(Date.parse(params['end_date']))
+        : new Date(new Date().setMonth(new Date().getMonth() + 1));
+    });
+
+    const today = new Date();
+    if (this.startDate.getTime() < today.setHours(0, 0, 0, 0)) {
+      this.page.params.ascending = 'false';
+    }
+
+    let paginationParams = this.page.params;
+    paginationParams.range_start = this.startDate.toLocaleString('en-GB');
+    paginationParams.range_end = this.endDate.toLocaleString('en-GB');
+    this.eventService.list(paginationParams).subscribe((page) => {
+      this.eventsPerDay = this.eventService.groupEventsByDate(page.items);
+    });
+
+    let prevUrl = '';
+    this.routeSubscription = this.router.events
+      .pipe(
+        filter((e) => e instanceof ActivationEnd),
+        distinctUntilChanged(() => this.router.url === prevUrl),
+        tap(() => (prevUrl = this.router.url))
+      )
+      .subscribe((_) => {
+        this.page.params.ascending = (
+          this.startDate.getTime() > today.setHours(0, 0, 0, 0)
+        ).toString();
+        let paginationParams = this.page.params;
+        paginationParams.range_start = this.startDate.toLocaleString('en-GB');
+        paginationParams.range_end = this.endDate.toLocaleString('en-GB');
+        this.eventService.list(paginationParams).subscribe((page) => {
+          this.eventsPerDay = this.eventService.groupEventsByDate(page.items);
+        });
+      });
+  }
```

Comment:
Nit: Consider refactoring this into its own function, like `configurePagination`, or instead provide a little bit more documentation (esp in the `routeSubscription` part!)